exercises/robot-simulator/src/example/kotlin/Robot.kt

```diff
@@ -0,0 +1,31 @@
+class Robot(var gridPosition: GridPosition = GridPosition(0, 0), var orientation: Orientation = Orientation.NORTH) {
+
+    fun advance() {
+        when (orientation) {
```

Comment:
This can be simplified a little by doing:

```
gridPosition = when(orientation) {
    Orientation.NORTH -> gridPosition.copy(y = gridPosition.y + 1)
    ...
}
```